usecases/traverser/explorer.go

```
@@ -212,6 +218,173 @@ func (e *Explorer) getClassVectorSearch(ctx context.Context,
+					if err != nil {
+						return nil, err
+					}
+
+					// Set the scoreexplain property to bm25 for every result
+					for i := range res1 {
+						scStr := res1[i].AdditionalProperties["score"].(string)
+						sc, _ := strconv.ParseFloat(scStr, 64)
+						res1[i].Score = float32(sc)
+						res1[i].ExplainScore = "(bm25)" + res1[i].ExplainScore
+					}
+
+					results = append(results, res1)
+				case "nearText":
+					sp := subsearch.SearchParams.(searchparams.NearTextParams)
+					weights = append(weights, subsearch.Weight)
+					if e.modulesProvider == nil {
+						continue
+					}
+
+					var err error
+					vector, err = e.modulesProvider.VectorFromInput(ctx,
+						params.ClassName, sp.Values[0])
+					if err != nil {
+						return nil, err
+					}
+					fmt.Printf("found vector: %v\n", shortenVectorString(10, vector))
+
+					res2, err := e.search.ClassVectorSearch(ctx, params.ClassName, vector, 0, 1000, nil)
+					if err != nil {
+						return nil, err
+					}
+
+					// Set the scoreexplain property to vector for every result
+					for i := range res2 {
+						res2[i].ExplainScore = fmt.Sprintf("(vector) %v %v ", shortenVectorString(10, vector), res2[i].ExplainScore)
+					}
+					results = append(results, res2)
+
+				case "nearVector":
+
+					sp := subsearch.SearchParams.(searchparams.NearVector)
+					weights = append(weights, subsearch.Weight)
+					if sp.Vector != nil && len(sp.Vector) != 0 {
+						vector = sp.Vector
+					}
+
+					res2, err := e.search.ClassVectorSearch(ctx, params.ClassName, vector, 0, 1000, nil)
+					if err != nil {
+						return nil, err
+					}
+
+					// Set the scoreexplain property to vector for every result
+					for i := range res2 {
+						res2[i].ExplainScore = fmt.Sprintf("(vector) %v %v ", shortenVectorString(10, vector), res2[i].ExplainScore)
+					}
+
+					results = append(results, res2)
+
+				default:
+					panic("unknown subsearch type:" + subsearch.Type)
```

Why panic instead of  just returning an error?
Panic might have sever impact on the application itself.